app/models/linkset.rb

```diff
@@ -4,11 +4,11 @@ class Linkset < ActiveRecord::Base
   LINKS = %w(home code docs wiki mail bugs).freeze
 
   LINKS.each do |url|
-    validates_formatting_of url.to_sym,
-      using: :url,
-      allow_nil: true,
-      allow_blank: true,
-      message: "does not appear to be a valid URL"
+    validates url.to_sym,
+      format: { with: %r{\Ahttps?:\/\/([^\s:@]+:[^\s:@]*@)?[A-Za-z\d\-]+(\.[A-Za-z\d\-]+)+\.?(:\d{1,5})?([\/?]\S*)?\z}i,
```

Comment:
Can we at least wrap this regexp into some constant and use it twice instead of specifying it twice in models?